lasagne/tests/layers/test_conv.py

```diff
@@ -181,20 +185,42 @@ def _convert(input, kernel, output, kwargs):
                                                        'stride': stride,
                                                        'flip_filters': True})
 
+    # extend (w/ and w/out symbolic output shape)
+    for symbolic in [False, True]:
+        input_shape = (4, 3, 7, 9)
+        input = np.random.random(input_shape)
+        kernel = np.random.random((16, 3, 2, 3))
+        stride = np.random.randint(2, 3, 2)
+        extend = [np.random.randint(0, s-1) for s in stride]
+        output = transposed_convNd(input, kernel, 0, stride, extend=extend)
+        kwargs = {'stride': stride, 'flip_filters': True}
+        if symbolic:
+            kwargs['output_size'] = theano.shared(np.array(output.shape[2:]))
+        else:
+            kwargs['output_size'] = np.array(output.shape[2:])
+        yield _convert(input, kernel, output, kwargs)
+
     # bias-less case
+    input_shape = (3, 1, 6, 13)
```

Comment:
If you move your `extend` test to the end of the function, you don't need to redefine the input shape here. That way, existing tests will stay the same as before.